### PR TITLE
16S metagenomics Nanopore training

### PR DESCRIPTION
**Pull request by gallardoalba**
16S metagenomics analysis of Nanopore data training, versión 1.0.

### CODE REVIEW

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+
+# Assign taxonomic classifications
+
+One of the key steps in metagenomic data analysis is to identify the taxon to which the individual reads belong. Taxonomic classification tools are based on microbial genome databases to identify the origin of each sequence.
+
+## Taxonomic classification with Kraken2
+
+In order to perform the taxonomic classification we will use __Kraken2__ ({% cite Wood_2019 %}), a tool which assign labels by examining the k-mers within a sequence and querying them in a database that uses a compact hash table for algorithm efficiency ({% cite deKoning2020 %}).
+
+![Taxonomic classification](../../images/metagenomics-nanopore/kmers-kraken.jpg "Kramer2 sequence classification algoritm ({% cite Wood2014 %})")
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:262]:
```suggestion
![Taxonomic classification](../../images/metagenomics-nanopore/kmers-kraken.jpg "Kramer2 sequence classification algorithm ({% cite Wood2014 %})")
```

Comment (beatrizserrano) [reply topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:262]: what is Kramer2?

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+For this tutorial we will use the __SILVA database__ ({% cite Quast2012 %}). It includes over 3.2 million 16S rRNA sequences from the _Bacteria_, _Archaea_ and _Eukaryota_ domains.
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:264]:
```suggestion
For this tutorial, we will use the __SILVA database__ ({% cite Quast2012 %}). It includes over 3.2 million 16S rRNA sequences from the _Bacteria_, _Archaea_ and _Eukaryota_ domains.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+
+> ### {% icon hands_on %} Hands-on: Task description
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:266]: task...

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+>    - *"Print scientific names instead of just taxids"*: `Yes`
+>    - *"Confidence"*: `0.1`
+>    - In *"Create Report"*:
+>        - *"Print a report with aggregrate counts/clade to file"*: `Yes`
+>        - *"Format report output like Kraken 1's kraken-mpa-report"*: `Yes`
+>    - *"Select a Kraken2 database"*: `Silva (Created: 2020-06-24T164526Z, kmer-len=35, minimizer-len=31, minimizer-spaces=6)`
+>
+>    > ### {% icon comment %} Comment
+>    >
+>    > A confidence score of 0.1 means that at least 10% of  kmers should match with de databse.  This value can be reduced if a less restrictive taxonomic assignation is desired.
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:280]:
```suggestion
>    > A confidence score of 0.1 means that at least 10% of the k-mers should match entries in the database. This value can be reduced if a less restrictive taxonomic assignation is desired. 
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+>    >
+>    > A confidence score of 0.1 means that at least 10% of  kmers should match with de databse.  This value can be reduced if a less restrictive taxonomic assignation is desired. 
+>    {: .comment}
+>
+{: .hands_on}
+
+
+> ### {% icon question %} Questions
+>
+> What does the term k-mer mean?
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:288]: isn't it too late for this? I have been wondering what a k-mer is for 20 lines or so ;)

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+> 
+>
+> > ### {% icon solution %} Solution
+> >
+> >  The term k-mer refers to all of a sequence's subsequences of length _k_, such that the sequence GATACA would have six monomers (G,A,T,A,C,A), five 2-mers (GA,AT,TA,AC,CA), four 3-mers (GAT,ATA,TAC,ACA), three 4-mer (GATA,ATAC,TACA) and two 5-mer (GATAC,ATACA).
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:293]:
```suggestion
> >  The term k-mer refers to all the sequence's subsequences of length _k_. For example, the sequence `GATACA` would have six monomers (`G`,`A`,`T`,`A`,`C`,`A`), five 2-mers (`GA`,`AT`,`TA`,`AC`,`CA`), four 3-mers (`GAT`,`ATA`,`TAC`,`ACA`), three 4-mer (`GATA`,`ATAC`,`TACA`) and two 5-mer (`GATAC`,`ATACA`).
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+> >
+> {: .solution}
+>
+{: .question}
+
+
+# Analyze taxonomic assigment
+
+Once we have assigned the corresponding taxa to each sequence, the next step is to be able to properly visualize the data, for which we will use the __Krona__ tool ({% cite Ondov_2011 %}). Previosuly it is necessary to adjust the format of the data output from __Kraken2__.
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:302]:
```suggestion
Once we have assigned the corresponding taxa to each sequence, the next step is to properly visualize the data, for which we will use the __Krona__ tool ({% cite Ondov_2011 %}). But before that, we need to adjust the format of the data output from __Kraken2__.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+
+> ### {% icon hands_on %} Hands-on: Task description
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:304]: task :)

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+>            - *"Replace with"*: `\t`
+>
+> 3. {% tool [Remove beginning](Remove beginning1) %} with the following parameters:
+>    - {% icon param-collection %} *"Dataset collection"*: `Replace Text on collection`
+>
+{: .hands_on}
+
+## Visualize the taxonomical classification with Krona
+
+__Krona__ allows hierarchical data to be explored with zooming, multi-layered pie charts. Through this tool we can easily visualize the composition of the bacterial communities, as well as compare how the populations of microorganisms are modified according to the conditions of the environment.
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:323]:
```suggestion
__Krona__ allows hierarchical data to be explored with zooming, multi-layered pie charts. With this tool, we can easily visualize the composition of the bacterial communities and compare how the populations of microorganisms are modified according to the conditions of the environment.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+{: .hands_on}
+
+Let's take a look at the result. Using the search bar we can check if certain taxa are present.
+
+<iframe id="krona" src="krona_all.html" frameBorder="0" width="100%" height="900px"> ![Krona](../../images/metagenomics-nanopore/krona.png) </iframe>
+
+> ### {% icon question %} Questions
+>
+> 1. What can we say about the health status of the soil samples?
+> 2. Are there significant differences in bacterial population structure between soil samples and rhizosphere samples?
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:341]:
```suggestion
> 2. Are there significant differences in the bacterial population structure between soil samples and rhizosphere samples?
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+>
+> > ### {% icon solution %} Solution
+> >
+> > 1. The low presence of _Alphaproteobacteria_ (including members of the order _Rhizobiales_), as well as the abundance of _Bacteroidetes_ and _Gammaproteobacteria_, suggest that the soil is highly exposed to phosphorus. This mineral is strongly associated with anthropogenic activity as it is an important component of agrochemicals.
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:345]:
```suggestion
> > 1. The low presence of _Alphaproteobacteria_ (including members of the order _Rhizobiales_), as well as the abundance of _Bacteroidetes_ and _Gammaproteobacteria_, suggest that the soil is highly exposed to phosphorus. This mineral is strongly associated with the anthropogenic activity as it is an important component of agrochemicals.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 12]
```diff
@@ -0,0 +1,355 @@
...
+> > 2. We can observe that there are important differences in the structure of the bacterial populations between the soil and rhizosphere samples. Particularly significant is the increase in phylum _Plactomycetes_, which are usually abundant in the rhizosphere.
+> >
+> {: .solution}
+>
+{: .question}
+
+# Conclusion
+{:.no_toc}
+
+In this tutorial we used MinION Nanopore sequencing data to study the health status of soil samples and the structure of bacterial populations. The results indicate that the soil suffers some degree of erosion as a result of their exposure to agrochemicals. We have also been able to study how the composition of microbial communities is modified in presence of plant organisms.
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:355]:
```suggestion
In this tutorial, we used MinION Nanopore sequencing data to study the health status of soil samples and the structure of bacterial populations. The results indicate that the soil suffers some degree of erosion as a result of their exposure to agrochemicals. We have also been able to study how the composition of microbial communities is modified in presence of plant organisms.
```

**Review by beatrizserrano — COMMENTED** [review]
Nice tutorial @gallardoalba!

**Comment by beatrizserrano** [comment]
@annefou this is the tutorial we talked about yesterday :)

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 13]
```diff
@@ -0,0 +1,355 @@
+---
+layout: tutorial_hands_on
+
+title: 16s microbial analysis with nanopore data
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:4]:
```suggestion
title: 16S Microbial analysis with nanopore data
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 14]
```diff
@@ -0,0 +1,355 @@
...
+>      - *"Which tool was used generate logs?"*: `FastQC`
+>      - {% icon param-collection %} *"Dataset collection"*:
+>
+>		- `FastQC unprocessed: Raw`
+> 4. Click on the {% icon galaxy-eye %} (eye) icon and inspect the generated HTML file
+> 
+>
+{: .hands_on}
+
+__FastQC__ provides information on various parameters, such as the range of quality values across all bases at each position. __MultiQC__ allows summarizing the output of different outputs from __FastQC__.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:133]:
```suggestion
{% tool [FastQC](toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1) %} provides information on various parameters, such as the range of quality values across all bases at each position. {% tool [MultiQC](toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1) %} allows summarizing the output of different outputs from {% tool [FastQC](toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1) %}.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 15]
```diff
@@ -0,0 +1,355 @@
...
+
+![FastQC plots length](../../images/metagenomics-nanopore/fastqc_sequence_length_distribution_plot.png "Sequence length distribution")
+
+In the first place, we are going to analyse the sequence length distribution of the different datasets.
+
+> ### {% icon question %} Question
+>
+> 1. Can you explain the sequence length distribution plot?
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:141]:
```suggestion
> 1. Can you explain the sequence length distribution plot? (Figure 3)
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 16]
```diff
@@ -0,0 +1,355 @@
...
+> >
+> {: .solution}
+{: .question}
+
+
+![FastQC plots quality](../../images/metagenomics-nanopore/fastqc_per_base_sequence_quality_plot.png "Per base sequence quality")
+
+
+
+If we examine figure 3, we can see that up to 3000 bp the quality of our sequencing data is around a __Phred score__ of 12, which is a relatively low value compared to other sequencing technologies. This is explained because Nanopore reads poses __high error rates in the basecalled reads__ (10% as compared to 1% for Illumina). However, Nanopore sequencing generates __very long reads__ (in theory only limited by the mechanisms of extraction of the genetic material), enabling the sequencing of the complete 16S rRNA gene, which makes it possible to identify bacterial taxa at higher resolution.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:155]:
```suggestion
If we examine figure 4, we can see that up to 3000 bp the quality of our sequencing data is around a __Phred score__ of 12, which is a relatively low value compared to other sequencing technologies. This is explained because Nanopore reads poses __high error rates in the basecalled reads__ (10% as compared to 1% for Illumina). However, Nanopore sequencing generates __very long reads__ (in theory only limited by the mechanisms of extraction of the genetic material), enabling the sequencing of the complete 16S rRNA gene, which makes it possible to identify bacterial taxa at higher resolution.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 17]
```diff
@@ -0,0 +1,355 @@
...
+
+> ### {% icon details %} FASTQ sequence quality format
+>
+> FASTQ format is a text-based format for storing both a biological sequence and its corresponding quality scores. Both the sequence letter and the quality score are encoded with a single ASCII character for brevity. You can find more info in the [Wikipedia article](FASTQ format).
+>
+{: .details}
+
+## Quality control using FastQC and MultiQC
+
+__FastQC__ is one of the most widely used tools to __check the quality__ of the samples generated by High Throughput Sequencing (HTS) technologies.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:114]:
```suggestion
{% tool [FastQC](toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1) %} is one of the most widely used tools to __check the quality__ of the samples generated by High Throughput Sequencing (HTS) technologies. 
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 18]
```diff
@@ -0,0 +1,355 @@
...
+
+![FastQC plots quality](../../images/metagenomics-nanopore/fastqc_per_base_sequence_quality_plot.png "Per base sequence quality")
+
+
+
+If we examine figure 3, we can see that up to 3000 bp the quality of our sequencing data is around a __Phred score__ of 12, which is a relatively low value compared to other sequencing technologies. This is explained because Nanopore reads poses __high error rates in the basecalled reads__ (10% as compared to 1% for Illumina). However, Nanopore sequencing generates __very long reads__ (in theory only limited by the mechanisms of extraction of the genetic material), enabling the sequencing of the complete 16S rRNA gene, which makes it possible to identify bacterial taxa at higher resolution.
+
+![FasTQC plots CG](../../images/metagenomics-nanopore/fastqc_per_sequence_gc_content_plot.png "Per sequence GC content")
+
+Let's have a look at figure 4, which shows the content of the GC by sequence.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:159]:
```suggestion
Let's have a look at figure 5, which shows the content of the GC by sequence.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 19]
```diff
@@ -0,0 +1,355 @@
...
+{: .question}
+
+> ### {% icon comment %} Comments
+> For more information on the topic of quality control, please see our training materials [here](https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html).
+{: .comment}
+
+
+# Improve the dataset quality
+
+In order to improve the quality of our data, we will use two tools presented above, __porechop__ and __fastp__.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:181]:
```suggestion
In order to improve the quality of our data, we will use two tools presented above, {% tool [Porechop](toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3) %} and {% tool [fastp](toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0) %}.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 20]
```diff
@@ -0,0 +1,355 @@
...
+
+## Adapter and chimera removal with porechop
+
+Nanopore sequencing technology requires to ligate adapters to both ends of genomic material to facilitate the strand capture and loading of a processive enzyme at the 5'end, boosting the effectiveness of the sequencing process. Adapter sequences should be removed because they __can interfere with aligment of reads__ to 16S rRNA gene reference database, for which we will use the __porechop__ tool ({% cite rrwick2017 %}).
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:185]:
```suggestion
Nanopore sequencing technology requires to ligate adapters to both ends of genomic material to facilitate the strand capture and loading of a processive enzyme at the 5'end, boosting the effectiveness of the sequencing process. Adapter sequences should be removed because they __can interfere with aligment of reads__ to 16S rRNA gene reference database, for which we will use the {% tool [Porechop](toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3) %} tool ({% cite rrwick2017 %}).
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 21]
```diff
@@ -0,0 +1,355 @@
...
+
+> ### {% icon comment %} Comment
+> If you are interested in Nanopore sequecing technology, you can find more information in {% cite Jain2016 %}.
+> 
+{: .comment}
+
+
+On the other hand, __chimeric sequences__ are considered a contaminant and should be removed because they can result in artificial inflation of the microbial diversity. With __Porechop__ you can eliminate them.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:193]:
```suggestion
On the other hand, __chimeric sequences__ are considered a contaminant and should be removed because they can result in artificial inflation of the microbial diversity. With {% tool [Porechop](toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3) %} you can eliminate them. 
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 22]
```diff
@@ -0,0 +1,355 @@
...
+> 1. {% tool [Porechop](toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3) %} with the following parameters:
+>    - *"Input FASTA/FASTQ"*: {% icon param-collection %} `soil collection`
+>    - *"Output format for the reads"*: `fastq`
+> 2. Rename the output as `soil collection trimmed`
+>
+{: .hands_on}
+
+## Filter sequences with fastp
+
+To increase the specificity of the analysis, we will select the reads with lengths between 1000 bp and 2000 bp, which are more informative from a taxonomic point of view, because they include both preserved and hypervariable regions of the 16S rRNA gene. In addition, sequences will be filtered on a minimum average read quality score of 9, according to the recommendations from {% cite Nygaard2020 %}. This stage will be carried out through the use of __fastp__ ({% cite Chen_2018 %}), an open-source tool designed to process FASTQ files.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:206]: You're also filtering just because you would expect the sequenced 16s rRNA gene has that length right. So you can assume that sequences shorter or longer are wrong in the first place?

```suggestion
To increase the specificity of the analysis, we will select the reads with lengths between 1000 bp and 2000 bp, which are more informative from a taxonomic point of view, because they include both preserved and hypervariable regions of the 16S rRNA gene. In addition, sequences will be filtered on a minimum average read quality score of 9, according to the recommendations from {% cite Nygaard2020 %}. This stage will be carried out through the use of {% tool [fastp](toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0) %} ({% cite Chen_2018 %}), an open-source tool designed to process FASTQ files.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 23]
```diff
@@ -0,0 +1,355 @@
...
+>        - *"PolyG tail trimming"*: `Disable polyG tail trimming`
+>
+> 2. Rename the output as `soil collection processed`
+>
+>
+{: .hands_on}
+
+# Re-evaluate datasets quality
+
+After processing the sequences, we are going to analyze them again using __FastQC__ and __MultiQC__ to see if we have managed to correct the anomalies that we had detected.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:231]:
```suggestion
After processing the sequences, we are going to analyze them again using {% tool [FastQC](toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1) %} and {% tool [MultiQC](toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1) %} to see if we have managed to correct the anomalies that we had detected.
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 24]
```diff
@@ -0,0 +1,355 @@
...
+>      - *"Which tool was used generate logs?"*: `FastQC`
+>      - {% icon param-collection %} *"Dataset collection"*:
+>
+>		- `FastQC processed: Raw`
+> 4. Click on the {% icon galaxy-eye %} (eye) icon and inspect the generated HTML file
+> 
+>
+{: .hands_on}
+
+We can verify that after processing the samples, the GC content presents a unimodal distribution, which indicates that the anomalies in the sequences have been successfully eliminated.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:250]:
```suggestion
We can verify that after processing the samples, the GC content presents a unimodal distribution, which indicates that the anomalies in the sequences have been successfully eliminated (Figure 6).
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 25]
```diff
@@ -0,0 +1,355 @@
...
+>
+>		- `FastQC unprocessed: Raw`
+> 4. Click on the {% icon galaxy-eye %} (eye) icon and inspect the generated HTML file
+> 
+>
+{: .hands_on}
+
+__FastQC__ provides information on various parameters, such as the range of quality values across all bases at each position. __MultiQC__ allows summarizing the output of different outputs from __FastQC__.
+
+![FastQC plots length](../../images/metagenomics-nanopore/fastqc_sequence_length_distribution_plot.png "Sequence length distribution")
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:135]: Are the two samples supposed to have different colors in the plot, or why are there two yellow lines?

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 26]
```diff
@@ -0,0 +1,355 @@
...
+
+# Assign taxonomic classifications
+
+One of the key steps in metagenomic data analysis is to identify the taxon to which the individual reads belong. Taxonomic classification tools are based on microbial genome databases to identify the origin of each sequence.
+
+## Taxonomic classification with Kraken2
+
+To perform the taxonomic classification we will use __Kraken2__ ({% cite Wood_2019 %}), a tool that assigns labels by examining the k-mers within a sequence and querying them in a database that uses a compact hash table for algorithm efficiency ({% cite deKoning2020 %}).
+
+![Taxonomic classification](../../images/metagenomics-nanopore/kmers-kraken.jpg "Kramer2 sequence classification algorithm ({% cite Wood2014 %})")
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:262]: I would say something about the algorithm if you want to include this figure.

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 27]
```diff
@@ -0,0 +1,355 @@
...
+> >  The term k-mer refers to all the sequence's subsequences of length _k_. For example, the sequence `GATACA` would have six monomers (`G`,`A`,`T`,`A`,`C`,`A`), five 2-mers (`GA`,`AT`,`TA`,`AC`,`CA`), four 3-mers (`GAT`,`ATA`,`TAC`,`ACA`), three 4-mer (`GATA`,`ATAC`,`TACA`) and two 5-mer (`GATAC`,`ATACA`).
+> >
+> {: .solution}
+>
+{: .question}
+
+
+# Analyze taxonomic assigment
+
+Once we have assigned the corresponding taxa to each sequence, the next step is to properly visualize the data, for which we will use the __Krona__ tool ({% cite Ondov_2011 %}). But before that, we need to adjust the format of the data output from __Kraken2__.
```
Comment (willemdek11) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:302]:
```suggestion
Once we have assigned the corresponding taxa to each sequence, the next step is to properly visualize the data, for which we will use the {% tool [Krona pie chart](toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1) %} tool ({% cite Ondov_2011 %}). But before that, we need to adjust the format of the data output from {% tool [Kraken2](toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.0.8_beta+galaxy0) %}.
```

**Review by willemdek11 — COMMENTED** [review]
Great work! Your story makes sense and I like the tools you have chosen to use 👍

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 28]
```diff
@@ -0,0 +1,355 @@
...
+
+# Introduction
+{:.no_toc}
+
+<!-- This is a comment. -->
+
+
+__Healthy soils__ are an essential element in maintaining the planet’s ecological balance. That is why their protection must be considered a priority in order to guarantee the well-being of humanity. The alteration of microbial populations often precedes changes in the physical and chemical properties of soils, so monitoring their condition can serve to predict their future evolution, allowing to develop stategies to __mitigate ecosystem damage__. Some of the species considered as __key bioindicators__ of soil quality are those belonging to the genera _Rhizobium_ and _Nitrosomonas_, because they are very sensitive to agro-industrial compounds.
+
+![Bacterial bioindicators](../../images/metagenomics-nanopore/bioindicators.jpg "Relationship between bacterial community composition and soil parameters {% cite Hermans2016 %}.")
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md]: @beatrizserrano I included some details about it.

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 29]
```diff
@@ -283,23 +294,10 @@ For this tutorial, we will use the __SILVA database__ ({% cite Quast2012 %}). It
...
-> >
-> {: .solution}
->
-{: .question}
-
 
 # Analyze taxonomic assigment
 
-Once we have assigned the corresponding taxa to each sequence, the next step is to properly visualize the data, for which we will use the {% tool [Krona pie chart](toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1) %} tool ({% cite Ondov_2011 %}). But before that, we need to adjust the format of the data output from {% tool [Kraken2](toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.0.8_beta+galaxy0) %}.
+Once we have assigned the corresponding taxa to each sequence, the next step is to properly visualize the data, for which we will use the __Krona pie chart__ tool ({% cite Ondov_2011 %}). But before that, we need to adjust the format of the data output from __Kraken2__.
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:300]: why this change?

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md [hunk 30]
```diff
@@ -0,0 +1,355 @@
...
+> >
+> > 1. The low presence of _Alphaproteobacteria_ (including members of the order _Rhizobiales_), as well as the abundance of _Bacteroidetes_ and _Gammaproteobacteria_, suggest that the soil is highly exposed to phosphorus. This mineral is strongly associated with the anthropogenic activity as it is an important component of agrochemicals.
+> > 2. We can observe that there are important differences in the structure of the bacterial populations between the soil and rhizosphere samples. Particularly significant is the increase in phylum _Plactomycetes_, which are usually abundant in the rhizosphere.
+> >
+> {: .solution}
+>
+{: .question}
+
+# Conclusion
+{:.no_toc}
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:351]: Done :)

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md (hunk 25 above)
Comment (gallardoalba) [reply topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:140]: @willemdek11 Indeed there are four samples. The color coding of the plot denotes the quality scores; green is considered high, yellow medium, and red low quality score.

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md (hunk 26 above)
Comment (gallardoalba) [reply topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md]: @willemdek11 I included some information about that.

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md (hunk 29 above)
Comment (gallardoalba) [reply topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md:300]: I consider that to include that information every time makes reading difficult, and in addition, it is not the style of most of the training. This information is included in the hands_on block.

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/workflows/main_workflow.ga [hunk 31]
```diff
@@ -0,0 +1,481 @@
+{
+    "a_galaxy_workflow": "true",
+    "annotation": "",
```
Comment (bgruening) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/workflows/main_workflow.ga:3]: Your tests are failling @gallardoalba ...

```
Linting topics/metagenomics/tutorials/nanopore-16S-metagenomics/workflows/main_workflow.ga

	The tags attribute is missing. Please add:

	"tags": ["metagenomics"]

	The 'annotation' attribute is missing or incorrect. Please add: "annotation": "<title of tutorial>"
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/workflows/main_workflow.ga [hunk 32]
```diff
@@ -1,8 +1,8 @@
 {
     "a_galaxy_workflow": "true",
-    "annotation": "",
+    "annotation": "16S rRNA analisis with Nanopore reads",
     "format-version": "0.1",
-    "name": "16S rRNA analisis with nanopore reads",
+    "name": "Training: 16S rRNA Analisis with Nanopore Sequencing Reads",
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/workflows/main_workflow.ga:5]:
```suggestion
    "name": "Training: 16S rRNA Analysis with Nanopore Sequencing Reads",
```

File: topics/metagenomics/tutorials/nanopore-16S-metagenomics/workflows/main_workflow.ga [hunk 33]
```diff
@@ -1,8 +1,8 @@
 {
     "a_galaxy_workflow": "true",
-    "annotation": "",
+    "annotation": "16S rRNA analisis with Nanopore reads",
```
Comment (beatrizserrano) [inline topics/metagenomics/tutorials/nanopore-16S-metagenomics/workflows/main_workflow.ga:3]:
```suggestion
    "annotation": "16S rRNA analysis with Nanopore reads",
```

**Review by beatrizserrano — APPROVED** [review]

**Comment by bgruening** [comment]
@willemdek11 is that PR ok for you? Thanks a lot for your review!

**Comment by willemdek11** [comment]
> @willemdek11 is that PR ok for you? Thanks a lot for your review!

Yes it is okay for me! :) Thank you

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thank everyone! @gallardoalba thanks for your first big contribution to the Galaxy community I hope we see many more in the next years. Welcome!

**Comment by shiltemann** [comment]
This is awesome, thanks @gallardoalba!! and a big thanks for the great reviews @beatrizserrano  @willemdek11 !

@hexylena fixed the site building issues, so this tutorial is live now :tada: https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html
